File: static/app/classes/Event.ts
/**
 *  Class that represent the information related to MiausicBox users
 *  @class Event
 */
import { User } from './User';
import {Band} from "./Band";

export class Event {
 


  /* Attributes */
  private _name:string;
  private _date:Date;
  private _creator:User;
  private _description:string;
  private _bands:Band[];
  private _direction:string;

  /* Constructor */
  constructor(name:string, date:Date, creator:User, description:string, bands:Band[], direction:string) {
    this._name = name;
    this._date = date;
    this._creator = creator;
    this._description = description;
    this._bands = bands;
    this._direction=direction;
  }

  /* Getters & Setters */
  get name():string {
    return this._name;
  }

  get date():Date {
    return this._date;
  }

  get creator():User {
    return this._creator;
  }

  get description():string {
    return this._description;
  }

  get bands():Band[]{
    return this._bands;
  }

  get direction():string{
    return this._direction;
  }

  set name(name:string) {
    this._name = name;
  }

  set date(date:Date) {
    this._date = date;
  }

  set creator(creator:User) {
    this._creator = creator;
  }

  set description(description:string) {
    this._description = description;
  }

  set bands(value:Band[]){
    this._bands=value;
  }

  set direction(value:string){
    this._direction=value;
  }

  /** Return if an Event is equal to other
   *  @method equals
   *  @param {object} object
   */
  equals(object:any){
    //At ApiRest object, this comparaison should be the with both ids
    if (!(object instanceof Event)) {
      return false;
    } else {
      var event:Event = object;
      return (event.name == this.name &&
              event.date == this.date &&
              event.creator == this.creator);
    };
  }

}

File: src/main/java/giraffe/miausicbox/model/Event.java
package giraffe.miausicbox.model;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

import giraffe.miausicbox.user.User;

@Entity
public class Event {

	public interface Basic {}
	
	public interface Followers extends User.Basic {}
	
	public interface Bands extends Band.Basic {}

	// Attributes
	@JsonView(Basic.class)
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	
	@JsonView(Basic.class)
	private String name;
	
	@JsonView(Basic.class)
	private String date;
	
	@JsonView(User.Basic.class)
	@OneToOne
	private User creator;
	
	@JsonView(Basic.class)
	private String description;
	
	@JsonView(Bands.class)
	@ManyToMany
	private List<Band> bands = new ArrayList<>();

	@JsonView(Basic.class)
	private String direction;

	@JsonView(Followers.class)
	@ManyToMany
	private List<User> followers = new ArrayList<>();
	
	// Constructor
	public Event() {}
	
	public Event(
			String name,
			String date,
			User creator,
			String description,
			List<Band> bands,
			String direction,
			List<User> followers
			) {
		super();
		this.name = name;
		this.date = date;
		this.creator = creator;
		this.description = description;
		this.bands = bands;
		this.direction = direction;
		this.followers = followers;
	}

	// Getters & Setters
	public long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDate() {
		return date;
	}

	public void setString(String date) {
		this.date = date;
	}

	public User getCreator() {
		return creator;
	}

	public void setCreator(User creator) {
		this.creator = creator;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public List<Band> getBands() {
		return bands;
	}

	public void setBands(List<Band> bands) {
		this.bands = bands;
	}

	public String getDirection() {
		return direction;
	}

	public void setDirection(String direction) {
		this.direction = direction;
	}

	public List<User> getFollowers() {
		return followers;
	}

	public void setFollowers(List<User> followers) {
		this.followers = followers;
	}
	
	// ToString
	@Override
	public String toString() {
		return (this.getName() + " - " + this.getDate().toString());
	}
	
	// Equals
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final Event other = (Event) obj;
		if (!Objects.equals(this.getId(), other.getId())) {
			return false;
		}
		return (Objects.equals(this.getName(), other.getName())
				&& (Objects.equals(this.getDate(), other.getDate())));
	}
	
}
